src/components/Glp/GlpSwap.js

```diff
@@ -422,10 +432,50 @@ export default function GlpSwap(props) {
     totalTokenWeights,
   ]);
 
+  useEffect(() => {
+    const { operation } = params;
+    const pay = queryParams.get("pay");
+    const receive = queryParams.get("receive");
+
+    if (operation) {
+      setIsBuying(operation.toLowerCase() === "buy");
```

Comment:
let's also create enums for param keys and possible values to better readability